chain/client/src/chunk_distribution_network.rs

```diff
@@ -0,0 +1,505 @@
+//! Helper functions for obtaining chunks from the chunk_distribution_network.
+
+use borsh::BorshDeserialize;
+use near_async::messaging::Sender;
+use near_chain::{
+    blocks_delay_tracker::BlocksDelayTracker, chain::BlockMissingChunks,
+    orphan::OrphanMissingChunks,
+};
+use near_chain_configs::{ChunkDistributionNetworkConfig, ClientConfig};
+use near_chunks::adapter::ShardsManagerRequestFromClient;
+use near_primitives::{
+    hash::CryptoHash,
+    sharding::{PartialEncodedChunk, ShardChunkHeader},
+    static_clock::StaticClock,
+    types::{EpochId, ShardId},
+};
+use std::fmt;
+use tracing::error;
+
+/// Basic interface for the chunk distribution network.
+pub trait ChunkDistributionClient {
+    type Error;
+    type Response;
+
+    async fn lookup_chunk(
+        &self,
+        prev_hash: CryptoHash,
+        shard_id: ShardId,
+    ) -> Result<Option<PartialEncodedChunk>, Self::Error>;
+
+    async fn publish_chunk(
+        &mut self,
+        chunk: &PartialEncodedChunk,
+    ) -> Result<Self::Response, Self::Error>;
+}
+
+/// Helper struct for the Chunk Distribution Network Feature.
+#[derive(Debug, Clone)]
+pub struct ChunkDistributionNetwork {
+    pub client: reqwest::Client,
+    pub config: ChunkDistributionNetworkConfig,
```

Comment:
nit: would be nice to make those as private fields and expose underlying functionality as member functions, for example introduce `ChunkDistributionNetwork::enabled` to replace `chunk_distribution.config.enabled`